packages/app/src/specs/SpecsListBanners.vue

```diff
@@ -223,32 +240,41 @@ watch(
   },
 )
 
-const cloudData = computed(() => ([props.gql.cloudViewer, props.gql.cachedUser, props.gql.currentProject] as const))
-const bannerToShow = computed(() => {
+const { isAllowedFeature } = usePromptManager()
+
+const bannerStateToShow = computed(() => {
+  const cloudStatus = userProjectStatusStore.cloudStatus
+  const projectStatus = userProjectStatusStore.projectStatus
+
+  if (cloudStatus !== 'allTasksCompleted' && isAllowedFeature('specsListBanner', cloudStatus)) {
+    return cloudStatus
+  }
+
+  if (projectStatus !== 'allTasksCompleted' && isAllowedFeature('specsListBanner', projectStatus)) {
+    return projectStatus
+  }
+
+  return null
+})
```

Comment:
I was really struggling with how to handle `usePromptManager` vs `isAllowedFeature` and what belonged where - the architecture wasn't really jiving with how I think about this sort of thing. The logic to determine _what_ to show lives in one place, but the logic to determine _whether it can be shown_ is in another - this works just fine provided there's one and only one flow of banners, but in this case we're adding a second flow that is blocked by the first until the first is done _or_ has been dismissed, so we can't just rely on `lastShown` or `dismissed` states.

I'll take another look to see if I can pull the status back inside - I agree it should be encapsulated, but I expect there will still end up being _some_ logic in `SpecsListBanners` to handle the fact that we're trying to juggle two different sets of banners (in addition to the many other banners here that aren't tied in to the manager)

Reply:
Discussed this with @warrensplayer - ended up pulling the `status` determination into `usePromptManager` since that's a better home for mapping a "Feature" to an applicable "banner state".